src/upstream_patches_ui.sh

```diff
@@ -305,15 +317,24 @@ function show_series_details()
 # @message_box: Text description to the list of patches
 # @_target_array_list: List of patches to be displayed
 # @previous_screen: Determines the next screen if an option is chosen
+# @fallback_message: Message for when there are no patches to display
 function list_patches()
 {
   local menu_title="$1"
   local -n _target_array_list="$2"
   local previous_screen="$3"
+  local fallback_message="$4"
   local selected_patch
   local ret
 
-  create_menu_options "${menu_title}" '' '_target_array_list' 1
+  if [[ -z "${_target_array_list}" ]]; then
+    create_message_box "${menu_title}" "${fallback_message}"
+    ret="$?"
+    screen_sequence['SHOW_SCREEN']='dashboard'
+    return "$ret"
```

Comment:
Move `secreen_sequence`  right after the `if` condition, and remove the `ret` variable part in favor of just returning `$?`.

Reply:
Sorry, I didn't understand the move of `screen_sequence`. When `list_patches` doesn't have patches to list, after the notification screen, the next screen is Dashboard. Otherwise, the next screen will be determined in the switch-case below. Moving `screen_sequence` after the `if` wouldn't be redundant?